signer/genericrsa/README.rst

```diff
@@ -0,0 +1,106 @@
+RSA Signing
+===============
+
+.. sectnum::
+.. contents:: Table of Contents
+
+This signer implements RSA PSS and PKCS1.5 signing. It accepts SHA1
+and SHA256 hashes on `/sign/hash` and data to be hashed on `/sign/data`.
+
+Example Usage:
+
+.. code:: bash
+
+    # hash your input data
+    $ echo foo | sha256sum -b | cut -d ' ' -f 1 | xxd -r -p | base64
+    tbudgBSg+bHWHiHnlteNzN8TUvI80ygS9IULh4rklEw=
+
+    # request a signature using the autograph client
+    $ go run client.go -D -a "tbudgBSg+bHWHiHnlteNzN8TUvI80ygS9IULh4rklEw=" \
+      -k dummyrsa -o /tmp/sig.bin -ko /tmp/pub.key
+
+    # format /tmp/pub.key to PEM (fold lines to 64 and add header and footer)
+    $ (echo '-----BEGIN PUBLIC KEY-----'; cat /tmp/pub.key |fold -w 64; echo;echo '-----END PUBLIC KEY-----') > /tmp/pub.pem
+
+    -----BEGIN PUBLIC KEY-----
+    MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAtEM/Vdfd4Vl9wmeVdCYu
+    WYnQl0Zc9RW5hLE4hFA+c277qanE8XCK+ap/c5so87XngLLfacB3zZhGxIOut/4S
+    lEBOAUmVNCfnTO+YkRk3A8OyJ4XNqdn+/ov78ZbssGf+0zws2BcwZYwhtuTvro3y
+    i62FQ7T1TpT5VjljH7sHW/iZnS/RKiY4DwqAN799gkB+Gwovtroabh2w5OX0P+PY
+    yUbJLFQeo5uiAQ8cAXTlHqCkj11GYgU4ttVDuFGotKRyaRn1F+yKxE4LQcAULx7s
+    0KzvS35mNU+MoywLWjy9a4TcjK0nq+BjspKX4UkNwVstvH18hQWun7E+dxTi59cR
+    mwIDAQAB
+    -----END PUBLIC KEY-----
+
+    # verify the signature with openssl
+    $ openssl pkeyutl -verify \
+    -in /tmp/inputhash.bin \
+    -sigfile /tmp/sig.bin \
+    -inkey /tmp/pub.pem -pubin -keyform PEM \
+    -pkeyopt rsa_padding_mode:pss -pkeyopt rsa_pss_saltlen:-1 -pkeyopt digest:sha1
+
+    Signature Verified Successfully
+
+Configuration
+-------------
+
+Requires PEM encoded public and private RSA keys.
+
+NB: if the publickey does not match the private key the monitor will
+break.
+
+For example:
+
+.. code:: yaml
+
+    signers:
+    - id: some-rsa-key
+      type: rsa
+      mode: pss
+      hash: sha256
+      saltlength: -1
+      privatekey: |
+        -----BEGIN RSA PRIVATE KEY-----
+        MIIEpAIBAAKCAQEAtEM/Vdfd4Vl9wmeVdCYuWYnQl0Zc9RW5hLE4hFA+c277qanE
+        ...
+        TDd4Me4PP+sTZeJ3RKvArDiMzEncDeMGZZnd4dBdi3LjzCNGTANAGw==
+        -----END RSA PRIVATE KEY-----
+      publickey: |
+        -----BEGIN PUBLIC KEY-----
+        MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAtEM/Vdfd4Vl9wmeVdCYu
+        ...
+        mwIDAQAB
+        -----END PUBLIC KEY-----
+
+Where:
+
+* hash is either sha1 or sha256
+* saltlength is only used in pss mode and is an integer
+  that correspond to https://golang.org/pkg/crypto/rsa/#pkg-constants
```

Comment:
I see that this is takes as a numeric byte padding length too, but I'd argue unless something uses that the string is preferable.

Reply:
Ideally we'd have an `Option<string, uint>` type in golang.

Reply:
not setting the value, or setting it to zero, means `auto` which I think is fine as a default. I can't foresee setting the length exactly but since the option allows it I don't want to limit that in config.

at the end of the day, regarding of the value, people will have to read the doc to figure out how to set it 🤷‍♂